optuna/integration/cma.py

```diff
@@ -310,17 +310,17 @@ def __init__(
                 highs.append(len(dist.choices) - 0.5)
             elif isinstance(dist, UniformDistribution) or isinstance(dist, LogUniformDistribution):
                 lows.append(self._to_cma_params(search_space, param_name, dist.low))
-                highs.append(self._to_cma_params(search_space, param_name, dist.high))
+                highs.append(self._to_cma_params(search_space, param_name, dist.high) - _EPS)
             elif isinstance(dist, DiscreteUniformDistribution):
                 r = dist.high - dist.low
                 lows.append(0 - 0.5 * dist.q)
-                highs.append(r + 0.5 * dist.q)
+                highs.append(r + 0.5 * dist.q - _EPS)
             elif isinstance(dist, IntUniformDistribution):
                 lows.append(dist.low - 0.5 * dist.step)
-                highs.append(dist.high + 0.5 * dist.step)
+                highs.append(dist.high + 0.5 * dist.step - _EPS)
```

Comment:
Is it required for ints as well where the upper bound is inclusive?

Reply:
Thank you for your review! As you pointed, this patch is not required for integers (and discrete uniforms) because these distributions accepts upper bound. Just to be sure, I executed following example but found no issues.

```python
import optuna


def objective(trial):
    x = trial.suggest_int("x", 0, 2)
    y = trial.suggest_discrete_uniform("y", 0, 2, 0.5)
    return x + y


if __name__ == "__main__":
    optuna.logging.set_verbosity(optuna.logging.WARNING)
    study = optuna.create_study(
        direction="maximize",
        sampler=optuna.samplers.CmaEsSampler())
    study.optimize(objective, timeout=20*60)
    print("Best value: {} (params: {})\n".format(study.best_value, study.best_params))
```


I'll fix it 👍

Reply:
I pushed f355888. PTAL.